### DISCUSSION TITLE
why doesn't setting `RIPGREP_CONFIG_PATH` cause ripgrep to use my config file?

### DISCUSSION THREAD

**Discussion by minusfive** [BurntSushi/ripgrep#2890] [Q&A] [answered]
Even though I have the `--hidden` flag set on my config file, `rg PATTERN` command doesn't honor it by default. Running `rg --hidden PATTERN` works as expected.

```sh
> /bin/cat $RIPGREP_CONFIG_PATH
# Include hidden files and directories
--hidden

# Ignore .git directory
--glob=!.git/*

# Automatically determine case-sensitivity
--smart-case
```

Every run with `--debug` always outputs `rg: DEBUG|rg::flags::parse|crates/core/flags/parse.rs:97: no extra arguments found from configuration file` as the first line.

```sh
> rg --version
ripgrep 14.1.1

features:+pcre2
simd(compile):+NEON
simd(runtime):+NEON

PCRE2 10.43 is available (JIT is available)
```

Any idea why that might be?

**Comment by BurntSushi** [comment, marked as answer]
Can you show the full `--debug` output?

My guess is that you aren't setting the environment variable correctly. This varies depending on your shell environment. The ripgrep docs assume you know how to set environment variables in a way that is carried to subprocesses executed from your shell. For example:

```
$ env | rg FOOBAR
$ echo $FOOBAR

$ FOOBAR=1
$ env | rg FOOBAR
$
$ echo $FOOBAR
1
```

Notice how `env` doesn't show `FOOBAR` as defined even though `echo $FOOBAR` does work. In order to make environment variables visible to subprocesses in a standard Unix shell, you need to export them:

```
$ export FOOBAR=1
$ env | rg FOOBAR
FOOBAR=1
$ echo $FOOBAR
1
```

**Reply by minusfive** [reply]
Duh! Sorry about that, I actually did know about `export` but forgot to do it here. So I guess it's the same as not knowing, amirite? It's been a while since I tweak my shell config. Apologies for the distraction.

Works fine now.
